### PR TITLE
Refresh `databricks_grants` with latest permissible grants

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
Refresh `databricks_grants` with latest permissible grants

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] ~covered with integration tests in `internal/acceptance`~
- [ ] relevant acceptance tests are passing
- [ ] ~using Go SDK~

### CODE REVIEW

File: catalog/resource_grants.go [hunk 1]
```diff
@@ -102,6 +102,9 @@ func getPermissionEndpoint(securable, name string) string {
 	if securable == "share" {
 		return fmt.Sprintf("/unity-catalog/shares/%s/permissions", name)
 	}
+	if securable == "the_connection" {
```
Comment (alexott) [inline catalog/resource_grants.go:105]: is `the_connection` correct name for securable?

Comment (nkvuong) [reply catalog/resource_grants.go]: typo 😶‍🌫️

Comment (nkvuong) [reply catalog/resource_grants.go]: I've added a test to confirm

**Review by mgyucht — COMMENTED** [review]
BROWSE, CREATE_MODEL, CREATE_FOREIGN_SCHEMA, and CREATE_FOREIGN_TABLE are not checked into OpenAPI. Where did these come from?

**Comment by nkvuong** [comment]
@mgyucht some of those permissions are in private preview, so they are in the UC service but not in OpenAPI - we need to discuss how to handle this better going forward, e.g. leave it to UC service to do the validation but that means tf plan won't show errors.

**Comment by mgyucht** [comment]
Understood. Sounds like we'll need a way to integrate it into the OpenAPI spec somehow.

**Review by mgyucht — APPROVED** [review]
LGTM!

**Pull request merged**